packages/contracts-bedrock/test/L2/SuperchainERC20.t.sol

```diff
@@ -65,159 +49,111 @@ contract SuperchainERC20Test is Test {
-        vm.prank(_caller);
-        superchainERC20.relayERC20(_caller, _to, _amount);
+        // Check the total supply and balance of `_to` after the mint were updated correctly
+        assertEq(superchainERC20.totalSupply(), _totalSupplyBefore + _amount);
+        assertEq(superchainERC20.balanceOf(_to), _toBalanceBefore + _amount);
     }
 
-    /// @notice Tests the `relayERC20` function reverts when the `crossDomainMessageSender` that sent the message is not
-    /// the same SuperchainERC20 address.
-    function testFuzz_relayERC20_notCrossDomainSender_reverts(
-        address _crossDomainMessageSender,
-        address _to,
+    /// @notice Tests the `burn` function reverts when the caller is not the bridge.
+    function testFuzz___superchainBurn_callerNotBridge_reverts(
+        address _caller,
+        address _from,
         uint256 _amount
     )
         public
     {
-        vm.assume(_to != ZERO_ADDRESS);
-        vm.assume(_crossDomainMessageSender != address(superchainERC20));
+        // Ensure the caller is not the bridge
+        vm.assume(_caller != SUPERCHAIN_ERC20_BRIDGE);
 
-        // Mock the call over the `crossDomainMessageSender` function setting a wrong sender
-        vm.mockCall(
-            MESSENGER,
-            abi.encodeWithSelector(IL2ToL2CrossDomainMessenger.crossDomainMessageSender.selector),
-            abi.encode(_crossDomainMessageSender)
-        );
+        // Expect the revert with `OnlySuperchainERC20Bridge` selector
+        vm.expectRevert(ISuperchainERC20Errors.OnlySuperchainERC20Bridge.selector);
 
-        // Expect the revert with `InvalidCrossDomainSender` selector
-        vm.expectRevert(ISuperchainERC20Errors.InvalidCrossDomainSender.selector);
+        // Call the `burn` function with the non-bridge caller
+        vm.prank(_caller);
+        superchainERC20.__superchainBurn(_from, _amount);
+    }
+
+    /// @notice Tests the `burn` function reverts when the amount is zero.
+    function testFuzz___superchainBurn_zeroAddressFrom_reverts(uint256 _amount) public {
+        // Expect the revert with `ZeroAddress` selector
+        vm.expectRevert(ISuperchainERC20Errors.ZeroAddress.selector);
 
-        // Call the `relayERC20` function with the sender caller
-        vm.prank(MESSENGER);
-        superchainERC20.relayERC20(_crossDomainMessageSender, _to, _amount);
+        // Call the `burn` function with the zero address
+        vm.prank(SUPERCHAIN_ERC20_BRIDGE);
+        superchainERC20.__superchainBurn({ _from: ZERO_ADDRESS, _amount: _amount });
     }
 
-    /// @notice Tests the `relayERC20` mints the proper amount and emits the `RelayERC20` event.
-    function testFuzz_relayERC20_succeeds(address _from, address _to, uint256 _amount, uint256 _source) public {
+    /// @notice Tests the `burn` burns the amount and emits the `Burn` event.
+    function testFuzz___superchainBurn_succeeds(address _from, uint256 _amount) public {
```

Comment:
you can add the new `SuperchainBurn` event emision check here. Same for `SuperchainMint` where it belongs